### PR TITLE
add AWSClusterTemplate type

### LINKED ISSUE #2575: Add AWSClusterTemplate type
/kind feature

**Describe the solution you'd like:**
Add a new CRD named AWSClusterTemplate. This is needed for ClusterClass.
In the short term this type will not be used outside of ClusterClass.

### PR DESCRIPTION
**Pull request by ykakarap**
**What type of PR is this?**
/kind feature
/kind api-change

**What this PR does / why we need it**:
This PR adds AWSClusterTemplate type. This type is need by the [ClusterClass](https://github.com/kubernetes-sigs/cluster-api/pull/4678) proposal.
In the near future this type would only be used by ClusterClass.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2575 

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Add AWSClusterTemplate type, to be used in managed cluster topologies (ClusterClass)
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -239,10 +239,10 @@ generate: ## Generate code
 
 .PHONY: generate-go
 generate-go: $(MOCKGEN)
-	go generate ./...
 	$(MAKE) generate-go-core
 	$(MAKE) generate-go-eks-bootstrap
 	$(MAKE) generate-go-eks-controlplane
+	go generate ./...
```
Comment (shivi28) [inline Makefile:245]: Is there any reason to change it?

**Comment by shivi28** [comment]
Thanks @ykakarap for the PR ,
Run `make generate` some tests are failing. It is always advisable to run it before raising any PR.

**Comment by ykakarap** [comment]
@shivi28 I did run `make generate` before submitting the PR. 
Similarly `make verify` runs successfully on my local. The same issue is surfacing on PRs for other providers too. 

The issue seems to be that the generated files on the pipeline are adding a extra `//go:build !ignore_autogenerated` at the top of the `zz_generate.deepcopy.go` files. 
This happened on rerunning the same pipeline on the same code that previously passed the verification pipeline. With this information I suspect that something about the pipelines changes and it might not be an issue with the code.

I dont know what is causing this, needs more investigation.

File: Makefile (hunk 1 above)
Comment (ykakarap) [reply Makefile:245]: this is to fix `make generate`.  Based on the [change](https://github.com/ykakarap/cluster-api-provider-aws/pull/1/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52) suggested by @randomvariable

**Comment by shivi28** [comment]
Ya even I tried `make generate` on this PR and nothing changed.
This looks like a CI issue only. Another thread going on for this discussion https://github.com/kubernetes-sigs/cluster-api/pull/4903#issuecomment-881212257

**Comment by ykakarap** [comment]
>This looks like a CI issue only.

Yes

File: Makefile (hunk 1 above)
Comment (randomvariable) [reply Makefile:245]: Yup.

**Comment by randomvariable** [comment]
See https://github.com/kubernetes-sigs/cluster-api/pull/4953

**Pull request merged**